### PR TITLE
Performance counters for nGraph

### PR DESCRIPTION
**Pull request by vladislav-volkov**
This fix adds new ITT performance counters to nGraph.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
BTW, is file size increased?

**Comment by vladislav-volkov** [comment]
> BTW, is file size increased?
I built the master and new-ngraph-perf-counters branches and got the following results:

/master/itt_off$ du
328900  ./Release/lib
336832  ./Release

/master/itt_on$ du
329772  ./Release/lib
337704  ./Release

/new-ngraph-perf-counters/itt_off$ du
328924  ./Release/lib
336856  ./Release

/new-ngraph-perf-counters/itt_on$ du
329804  ./Release/lib
337736  ./Release

File: ngraph/src/ngraph/itt.hpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ namespace ngraph
         namespace domains
         {
             OV_ITT_DOMAIN(Ngraph);
+            OV_ITT_DOMAIN(NgraphOp);
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply ngraph/src/ngraph/itt.hpp:33]: The separate domain is useful for filter out nGraph operations and we can collect statistics about operations.

[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/src/ngraph/itt.hpp:33]: In this case can we use the next name for domain: nGraph::Op?

Also can we rename `Ngraph` to `nGraph`? It is more aligned with our documentation :)

[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply ngraph/src/ngraph/itt.hpp:33]: Ok, l'll change.
To change nGraph::Op is little bit diffuclt but I'll try (:

[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/src/ngraph/itt.hpp:33]: `nGraph::Op` is just a proposal, if it is difficult we can use another name.

**Comment by vladislav-volkov** [comment]
Unfortunatelly on Linux* systems ittnotify requires dl an pthreads. ):

File: openvino/itt/cmake/FindITT.cmake [hunk 2]
```diff
@@ -47,10 +47,13 @@ find_package_handle_standard_args(ITT
...
 if(ITT_FOUND)
     set(INTEL_ITT_FOUND ${ITT_FOUND})
+
     add_library(ittnotify STATIC IMPORTED GLOBAL)
     set_target_properties(ittnotify PROPERTIES IMPORTED_LOCATION "${Located_ITT_LIBS}"
                                                INTERFACE_INCLUDE_DIRECTORIES ${Located_ITT_INCLUDE_DIRS}
                                                INTERFACE_COMPILE_DEFINITIONS ENABLE_PROFILING_ITT)
 
-    set(INTEL_ITT_LIBS ittnotify)
+    if(LINUX)
```
Comment (ilya-lavrenov) [inline openvino/itt/cmake/FindITT.cmake:56]: `LINUX` -> `UNIX`?

Comment (vladislav-volkov) [reply openvino/itt/cmake/FindITT.cmake]: fixed

**Pull request merged**